OloEngine/tests/Rendering/RenderingRegressionTest.cpp

```
@@ -0,0 +1,86 @@
+// =============================================================================
+
+TEST(RenderingRegression, WaterWavelengthMustBePositive)
+{
+    WaterComponent wc{};
+    EXPECT_GT(wc.m_Wavelength0, 0.0f)
+        << "Wavelength0 must be > 0 for visible waves";
+    EXPECT_GT(wc.m_Wavelength1, 0.0f)
+        << "Wavelength1 must be > 0 for visible waves";
+}
+
+TEST(RenderingRegression, WaterWavelengthCopied)
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)